src/java_bytecode/simple_method_stubbing.cpp

```diff
@@ -0,0 +1,298 @@
+  namespacet ns(symbol_table);
+
+  // synthetic source location that contains the opaque function name only
+  source_locationt synthesized_source_location;
+  synthesized_source_location.set_function(symbol.name);
+
+  // Initialize the return value or `this` parameter under construction.
+  // Note symbol.type is required_type, but with write access
+  // Probably required_type should not be const
+  const bool is_constructor = required_type.get_is_constructor();
+  if(is_constructor)
+  {
+    const auto &this_argument = required_type.parameters()[0];
+    const typet &this_type = this_argument.type();
+    symbolt &init_symbol = get_fresh_aux_symbol(
+      this_type,
+      "to_construct",
+      "to_construct",
+      synthesized_source_location,
+      ID_java,
+      symbol_table);
+    init_symbol.type = this_type;
+    const symbol_exprt &init_symbol_expression = init_symbol.symbol_expr();
+    code_assignt get_argument(
+      init_symbol_expression, symbol_exprt(this_argument.get_identifier()));
+    get_argument.add_source_location() = synthesized_source_location;
+    new_instructions.copy_to_operands(get_argument);
+    create_method_stub_at(
+      this_type,
+      init_symbol_expression,
+      synthesized_source_location,
+      new_instructions,
+      1,
+      true,
+      false);
+  }
+  else
+  {
+    const typet &required_return_type = required_type.return_type();
+    if(required_return_type != empty_typet())
+    {
+      symbolt &to_return_symbol = get_fresh_aux_symbol(
+        required_return_type,
+        "to_return",
+        "to_return",
+        synthesized_source_location,
+        ID_java,
+        symbol_table);
+      to_return_symbol.type = required_return_type;
+      const exprt &to_return = to_return_symbol.symbol_expr();
+      if(to_return_symbol.type.id()!=ID_pointer)
+      {
+        gen_nondet_init(
+          to_return,
+          new_instructions,
+          symbol_table,
+          source_locationt(),
+          false,
+          allocation_typet::LOCAL, // Irrelevant as type is primitive
+          !assume_non_null,
+          object_factory_parameters,
+          update_in_placet::NO_UPDATE_IN_PLACE);
+      }
+      else
+        create_method_stub_at(
+          required_return_type,
+          to_return,
+          synthesized_source_location,
+          new_instructions,
+          0,
+          false,
+          false);
+      new_instructions.copy_to_operands(code_returnt(to_return));
+    }
+  }
+
+  symbol.value = new_instructions;
+}
+
+/// Replaces `sym` with a function stub per the function above if it is
+///   of suitable type.
+/// \param symname: Symbol name to consider stubbing
+void java_simple_method_stubst::check_method_stub(
+  const irep_idt &symname)
+{
+  const symbolt &sym=*symbol_table.lookup(symname);
+  if(sym.is_type)
+    return;
+  if(sym.value.id()!=ID_nil)
+    return;
+  if(sym.type.id()!=ID_code)
+    return;
```

Comment:
Nit pick: the use of `&&` would make this code more concise and would express what is the condition instead of saying when not to do this. Like this:
`if(!sym.is_type && sym.value.id() == ID_nil && sym.type.id() == ID_code) create_method_stub(...);`  (while moving the two `sym.name` to the top to clearly mark the exceptional case.